js/components/main.js

```diff
@@ -233,88 +234,82 @@ class Main extends ImmutableComponent {
 
   registerSwipeListener () {
     // Navigates back/forward on macOS two-finger swipe
-    var trackingFingers = false
-    var swipeGesture = false
-    var isSwipeOnLeftEdge = false
-    var isSwipeOnRightEdge = false
-    var deltaX = 0
-    var deltaY = 0
-    var startTime = 0
-    var time
-
-    this.mainWindow.addEventListener('wheel', (e) => {
-      if (trackingFingers) {
-        deltaX = deltaX + e.deltaX
-        deltaY = deltaY + e.deltaY
-        time = (new Date()).getTime() - startTime
-      }
-    }, { passive: true })
-    ipc.on(messages.DEBUG_REACT_PROFILE, (e, args) => {
-      window.perf = require('react-addons-perf')
-      if (!window.perf.isRunning()) {
-        if (!window.isFirstProfiling) {
-          window.isFirstProfiling = true
-          console.info('See this blog post for more information on profiling: http://benchling.engineering/performance-engineering-with-react/')
-        }
-        currentWindow.openDevTools()
-        console.log('starting to profile...')
-        window.perf.start()
-      } else {
-        window.perf.stop()
-        console.log('profiling stopped. Wasted:')
-        window.perf.printWasted()
-      }
-    })
-    ipc.on(messages.OPEN_BRAVERY_PANEL, (e) => {
-      if (!this.braveShieldsDisabled) {
-        this.onBraveMenu()
-      } else {
-        windowActions.newFrame({
-          location: 'about:preferences#shields',
-          singleFrame: true
-        }, true)
-      }
-    })
+    let swipeGesture = false
+
     ipc.on(messages.ENABLE_SWIPE_GESTURE, (e) => {
       swipeGesture = true
     })
+
     ipc.on(messages.DISABLE_SWIPE_GESTURE, (e) => {
       swipeGesture = false
     })
-    ipc.on('scroll-touch-begin', function () {
-      if (swipeGesture &&
-        systemPreferences.isSwipeTrackingFromScrollEventsEnabled()) {
-        trackingFingers = true
-        startTime = (new Date()).getTime()
-      }
-    })
-    ipc.on('scroll-touch-end', function () {
-      if (time > 50 && trackingFingers && Math.abs(deltaY) < 50) {
-        if (deltaX > 70 && isSwipeOnRightEdge) {
-          ipc.emit(messages.SHORTCUT_ACTIVE_FRAME_FORWARD)
-        } else if (deltaX < -70 && isSwipeOnLeftEdge) {
-          ipc.emit(messages.SHORTCUT_ACTIVE_FRAME_BACK)
+
+    // isSwipeTrackingFromScrollEventsEnabled is only true if "two finger scroll to swipe" is enabled
+    // the swipe gesture handler will only fire if the three finger swipe setting is on, so the complete off setting and three and two finger together is also taken care of
+    if (systemPreferences.isSwipeTrackingFromScrollEventsEnabled()) {
```

Comment:
Could you put this check in `scroll-touch-begin`? Because we don't want user to restart for every system setting change like other browsers